system-test/logging-bunyan.ts

```diff
@@ -55,6 +57,38 @@ describe('LoggingBunyan', function () {
     assert.ok(loggingBunyan.cloudLog instanceof LogSync);
   });
 
+  it('should write diagnostic entry', async () => {
+    instrumentation.setInstrumentationStatus(false);
+    const start = Date.now();
+    logger.info(MESSAGE);
+    const entries = await pollLogs(
+      LOG_NAME,
+      start,
+      2,
+      WRITE_CONSISTENCY_DELAY_MS
+    );
+    assert.strictEqual(entries.length, 2);
+    entries.forEach(entry => {
+      assert.ok(entry.data);
+      if (
+        Object.prototype.hasOwnProperty.call(
+          entry.data,
+          instrumentation.DIAGNOSTIC_INFO_KEY
+        )
+      ) {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const info = (entry.data as any)[instrumentation.DIAGNOSTIC_INFO_KEY][
+          instrumentation.INSTRUMENTATION_SOURCE_KEY
+        ];
+        assert.equal(info[0].name, 'nodejs');
+        assert.equal(info[1].name, 'nodejs-bunyan');
+      } else {
+        const data = entry.data as {message: string};
+        assert.ok(data.message.includes(MESSAGE));
+      }
+    });
+  });
+
```

Comment:
nit: extra tests might be useful:
- when the instrumentation was already sent once (make sure it isn't sent again)
- getNodejsLibraryVersion function

Reply:
The test for sending instrumentation once is well covered in base nodejs-library. As for library version, I will add a check to see the library version present.